dbclient/health/src/main/java/io/helidon/dbclient/health/DbClientHealthCheck.java

```diff
@@ -58,39 +72,204 @@ public static Builder builder(DbClient dbClient) {
         return new Builder(dbClient);
     }
 
+    /**
+     * Execute the ping statement.
+     *
+     * @return {@code Awaitable} instance to wait for
+     */
+    protected abstract Awaitable<?> execPing();
+
     @Override
     public HealthCheckResponse call() {
-        HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
-                .name(name);
+        HealthCheckResponseBuilder builder = HealthCheckResponse.builder().name(name);
 
         try {
-            dbClient.ping().await(10, TimeUnit.SECONDS);
+            execPing().await(timeoutSeconds, TimeUnit.SECONDS);
             builder.up();
         } catch (Throwable e) {
             builder.down();
             builder.withData("ErrorMessage", e.getMessage());
             builder.withData("ErrorClass", e.getClass().getName());
-            e.printStackTrace();
+            // Do not process logging arguments until it makes sense.
+            if (LOGGER.isLoggable(Level.FINER)) {
+                LOGGER.log(Level.FINER, String.format(
+                        "Database %s is not responding: %s", dbClient.dbType(), e.getMessage()), e);
+            }
         }
 
         return builder.build();
     }
 
+    protected DbClient dbClient() {
+        return dbClient;
+    }
+
+    /**
+     * Database health check which calls default DBClient's {@code ping} method.
+     */
+    private static final class DbClientHealthCheckAsPing extends DbClientHealthCheck {
+
+        private DbClientHealthCheckAsPing(Builder builder) {
+            super(builder);
+            LOGGER.log(Level.FINEST, "Created an instance of DbClientHealthCheckAsPing");
+        }
+
+        @Override
+        protected Awaitable<Void> execPing() {
+            return dbClient().ping();
+        }
+
+    }
+
+    /**
+     * Database health check which calls DBClient's {@code namedDml} method.
+     */
+    private static final class DbClientHealthCheckAsNamedDml extends DbClientHealthCheck {
+
+        /** Name of the statement. */
```

Comment:
Please do not put Javadoc on private fields, use either single line or multi line comment instead.

Reply:
Changed all those somments to `/* ... */`